src/rules.jl

```diff
@@ -158,6 +158,20 @@ struct DNERule <: AbstractRule end
 
 DNERule(args...) = DNE()
 
+#####
+##### `NotImplementedRule`
+#####
+
+"""
+    NotImplementedRule <: AbstractRule
+
+Rule indicating that a particular derivative is not implemented by ChainRules.
```

Comment:
As with @oxinabox 's comment above, I would like to see this expanded slightly to say more about what a `NotImplementedRule` _does_ mean. Some context would be helpful e.g.
```julia
"""
Rule indicating that a particular derivative is not yet implemented by ChainRules. Different from [`DNERule`](@ref) in that this rule does not imply non-differentiability, but rather that no one has implemented it yet.
"""
```